src/backend/distributed/commands/table.c

```diff
@@ -162,31 +163,61 @@ PreprocessDropTableStmt(Node *node, const char *queryString,
 
 
 /*
- * PostprocessCreateTableStmt takes CreateStmt object as a parameter and errors
- * out if it creates a table with a foreign key that references to a citus local
- * table.
+ * PostprocessCreateTableStmt takes CreateStmt object as a parameter and
+ * processes foreign keys on relation via PostprocessCreateTableStmtForeignKeys
+ * function.
  *
  * This function also processes CREATE TABLE ... PARTITION OF statements via
  * PostprocessCreateTableStmtPartitionOf function.
  */
 void
 PostprocessCreateTableStmt(CreateStmt *createStatement, const char *queryString)
+{
+	PostprocessCreateTableStmtForeignKeys(createStatement);
+
+	if (createStatement->inhRelations != NIL && createStatement->partbound != NULL)
```

Comment:
is partbound also NULL for the default partition?

Reply:
not sure, just used the old code

Reply:
seems it would not be null